src/tabookey-gasless/RelayClient.js

```diff
@@ -399,7 +409,12 @@ class RelayClient {
     let pinger = await this.serverHelper.newActiveRelayPinger(blockFrom, gasPrice);
     let errors = [];
     for (;;) {
-      let activeRelay = await pinger.nextRelay();
+      let activeRelay
+      if (self.config.predefinedRelay) {
+        activeRelay = self.config.predefinedRelay;
```

Comment:
A small issue here is that if the predefined relayer is down for whatever reason, the provider will re-attempt with the same one over and over again. This is more of an underlying issue with this whole function, however. What sort of behavior would you expect in this scenario @pertsev?

Reply:
First of all, I think we should rename a predefined relayer to _preferred_ relayer as you have mentioned above.

And yes, the client should try to connect to other relayers if preferred fails. That would be more reliable I think.